homeassistant/components/rfxtrx/binary_sensor.py

```diff
@@ -85,6 +106,14 @@ async def async_setup_entry(
 
     discovery_info = config_entry.data
 
+    def get_sensor_description(type_string: str, device_class: str | None = None):
+        description = SENSOR_TYPES_DICT.get(type_string)
+        if description is None:
+            description = BinarySensorEntityDescription(key=type_string)
+        if device_class:
+            description = replace(description, device_class=device)
```

Comment:
`device` is not a valid device class. It also doesn't look as `device` is defined when we call `get_sensor_description`.

Reply:
Seems so. This code will go away once we drop yaml. Can't set this override from config flow. Seems we had no test for that case.

I'll fix.

Reply:
https://github.com/home-assistant/core/pull/54200